lib/XTR/MaintenanceRegularTask.php

```diff
@@ -523,4 +532,27 @@ private function assessDynamicDisplayGroups()
             $this->runMessage .= ' - Done (not required)' . PHP_EOL . PHP_EOL;
         }
     }
-}
\ No newline at end of file
+
+    private function tidyAdCampaignSchedules()
+    {
+        $this->runMessage .= '## ' . __('Tidy Ad Campaign Schedules') . PHP_EOL;
+        Profiler::start('RegularMaintenance::tidyAdCampaignSchedules', $this->log);
+        $count = 0;
+
+        foreach ($this->scheduleFactory->query(null, [
```

Comment:
Would it be better to make it possible to query the `parentCampaignId` in schedule factory